src/odemis/acq/test/flim_test.py

```diff
@@ -272,6 +279,101 @@ def test_setting_stream(self):
+        # Currently the implementation does the second option
+        helper.roi.value = (0.20, 0.20, 0.80, 0.80)
+        numpy.testing.assert_almost_equal(helper.roi.value, (0.20, 0.20, 0.80, 0.80))
+        self.assertEqual(helper.repetition.value, (256, 256))
+        self.assertLess(helper.pixelSize.value, pxs256f)
+
+        # TODO: reduce the ROI
+
+        pxs256h = helper.pixelSize.value
+
+        # Same thing but only change in one dimension (typical, from the GUI)
+        # X is twice smaller => exactly twice less rep, that's easy
+        helper.roi.value = (0.20, 0.20, 0.50, 0.80)
+        numpy.testing.assert_almost_equal(helper.roi.value, (0.20, 0.20, 0.50, 0.80))
+        self.assertEqual(helper.repetition.value, (128, 256))
+        self.assertAlmostEqual(helper.pixelSize.value, pxs256h)
+
+        # Slightly decrease X
+        # -> rep stays the same, pxs stays the same.
+        # ROI adjusted slightly on X (ideally it wouldn't), but size is the same
+        helper.roi.value = (0.20, 0.20, 0.45, 0.80)
+        self.assertEqual(helper.repetition.value, (128, 256))
+        self.assertAlmostEqual(helper.pixelSize.value, pxs256h)
+        roi = helper.roi.value
+        roi_size = (roi[2] - roi[0], roi[3] - roi[1])
+        numpy.testing.assert_almost_equal(roi_size, (0.3, 0.6))
+
+        # Move (back) => everything as requested
+        helper.roi.value = (0.20, 0.20, 0.50, 0.80)
+        numpy.testing.assert_almost_equal(helper.roi.value, (0.20, 0.20, 0.50, 0.80))
+
+        # Slightly increase X
+        # -> rep increases (on X), while ROI & pixel size slightly smaller
+        helper.roi.value = (0.20, 0.20, 0.60, 0.80)
+        self.assertEqual(helper.repetition.value, (256, 256))
+        self.assertAlmostEqual(helper.pixelSize.value, pxs256h)
+        roi = helper.roi.value
+        roi_size = (roi[2] - roi[0], roi[3] - roi[1])
+        numpy.testing.assert_almost_equal(roi_size, (0.6, 0.6))
+
+        helper.roi.value = (0.15, 0.6, 0.8, 0.8)
+        helper.repetition.value = (1, 1)
+
+    def test_setting_stream_small_roi(self):
+        """
+        Check the small ROI can be acquire with large rep (=> scale < 1)
+        """
+
+        helper = stream.ScannedTCSettingsStream('Stream', self.apd, self.ex_light, self.lscanner,
+                                                self.sft, self.tc_scanner)
+
+        # Ask full ROI -> should have full ROI
+        helper.roi.value = (0, 0, 1, 1)
+        helper.repetition.value = (2048, 2048)
+        numpy.testing.assert_almost_equal(helper.roi.value, (0, 0, 1, 1))
+        self.assertEqual(helper.repetition.value, (2048, 2048))
+
+        # quarter of the ROI, still should be able to do 2048x2048
+        helper.roi.value = (0.25, 0.25, 0.75, 0.75)
+        helper.repetition.value = (2048, 2048)
+        numpy.testing.assert_almost_equal(helper.roi.value, (0.25, 0.25, 0.75, 0.75))
+        self.assertEqual(helper.repetition.value, (2048, 2048))
+
```

Comment:
Add a test that checks the px size has been decreased accordingly (also about 1/4 of the previous value?)?